packages/app/lib/ui/widgets/pagination_used_for_tags_list.dart

```diff
@@ -0,0 +1,148 @@
+    this._initScrollController();
+  }
+
+  void _initScrollController() {
+    scrollController.addListener(() {
+      final _scrollOffset = scrollController.offset;
+      final _maxOffset = scrollController.position.maxScrollExtent;
+      if (_scrollOffset >= _maxOffset) {
+        if (this.widget.paginator.fetchMore != null) {
+          this.widget.paginator.fetchMore!();
+        }
+      }
+    });
+  }
+
+  Widget _error(BuildContext context, String errorMessage) {
+    return SingleChildScrollView(
+      child: ErrorCard(
+          message:
+              AppLocalizations.of(context)!.paginationListError(errorMessage)),
+    );
+  }
+
+  Widget _loading() {
+    return Center(
+        child: Container(
+            padding: EdgeInsets.all(15.0),
+            child: CircularProgressIndicator(
+              color: MeliColors.black,
+            )));
+  }
+
+  Widget _emptyResult(BuildContext context) {
+    return SingleChildScrollView(
+      child: Column(children: [
+        Text(AppLocalizations.of(context)!.paginationListNoResults,
+            style: TextStyle(fontStyle: FontStyle.italic)),
+      ]),
+    );
+  }
+
+
+  Widget _loadMore(BuildContext context, bool isLoading) {
+    return Column(children: [isLoading ? this._loading() : Text("...")]);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Query(
+      options:
+          QueryOptions(document: this.widget.paginator.nextPageQuery(null)),
+      builder: (result, {VoidCallback? refetch, FetchMore? fetchMore}) {
+        if (this.widget.paginator.refresh == null) {
+          // Workaround to access `refetch` method from the outside
+          this.widget.paginator.refresh = refetch;
+        }
+
+        if (result.hasException) {
+          return this._error(context, result.exception.toString());
+        }
+
+        if (result.isLoading && result.data == null) {
+          return this._loading();
+        }
+
+        final data = this
+            .widget
+            .paginator
+            .parseJSON(result.data as Map<String, dynamic>);
+
+        if (data.documents.isEmpty) {
+          return this._emptyResult(context);
+        }
+
+        FetchMoreOptions opts = FetchMoreOptions(
+          document: this.widget.paginator.nextPageQuery(data.endCursor),
+          updateQuery: (previousResultData, fetchMoreResultData) {
+            return this
+                .widget
+                .paginator
+                .mergeResponses(previousResultData!, fetchMoreResultData!);
+          },
+        );
+
+        // Get to access `fetchMore` method from the outside
+        this.widget.paginator.fetchMore = () {
+          if (data.hasNextPage) {
+            fetchMore!(opts);
+          }
+        };
+
+        // Deduplicate the returned collection.
+        var seen = Set<String>();
+        List<UsedFor> uniqueUses = data.documents
+            .where((usedFor) => seen.add(usedFor.usedFor))
+            .toList();
```

Comment:
In the next line I account for the situation you describe by re-requesting items until we get a certain amount of unique uses. (still a hack on a hack)...

https://github.com/p2panda/meli/blob/ac0ab64edc5107e46c58f1f95fe00d43bbaeb100/packages/app/lib/ui/widgets/pagination_used_for_tags_list.dart#L132-L137

But yeh, totally agree that if we re-think the data type design a bit we could really make this nicer. 

I think the a really nice way of modelling this would be to move the sighting relation field into it's own "join" document: 

```
BeeAttributeUsedFor {
  description: str
}

BeeAttributeAssociation {
  attribute: relation(bee_attribute_used_for)
  relation: relation(bee_sightning)
}
```

This way we get;
- better re-use of UsedFor attributes
- nice aggregates over UsedFor attributes (`totalCount`) 
- familiar pattern in the app for querying, displaying, auto-completing attributes (no dedup needed)
- if you delete an "association" it doesn't actually delete the attribute

Reply:
Any change like this we make we would want to do for the other attributes (location).

Reply:
We _could_ combine the attribute association for all attributes into one like so:

```
BeeAttributeAssociation {
  attribute: relation(bee_attribute_used_for)
  sighting: relation(bee_sightning)
}

```

Reply:
We end up with even more documents and schema :rofl: but maybe that's just unavoidable with any kind of meaningful application.

Reply:
Alternatively we do it all like `local_name` (relation list with empty list representing no value) and add attribute fields for every possible attribute to bee sighting. 

Here we also get all the nice repeatable behaviour around the attributes but with fewer schema (just a big sighting).

Reply:
> Any change like this we make we would want to do for the other attributes (location).

Wondering what data we're interested in re-using and which not? Or where do we need extra value from knowing that data appeared across sightings?

With "location" I think its fine as we've designed it, as it is really only an attribute of that specific sighting.

Reply:
Ah yeh, that's true, locations aren't reused in the same way, that's cool :+1:

Reply:
What do you think of adding the join document then? Seems quite worth it to me.

Reply:
> What do you think of adding the join document then? Seems quite worth it to me.

It's cool! I wouldn't introduce it just for "Used for" maybe? Or is there a reason this data behaves differently than "local name"? Maybe use the same pattern for both?

Reply:
I think the difference between "local_name" and "used_for" is that with the former we are only wanting one value per sighting, whereas with the later we want a list.

Reply:
Of course, "local_name" is actually a list 🙃 but in terms of how we allow users to interact with the data, this is the difference.

Reply:
Is this a fundamental difference wrt this conversation about the data design?... I'm not actually sure...

Reply:
We did talk about allowing "local_name" to be a list at one point, as different people might know the same bee by different local names.